### ISSUE TITLE
Setting the MapRenderer.Center too far from its current location leads to "System.ArgumentNullException: Buffer cannot be null."

### ISSUE THREAD

**Issue by FredMoreau** [microsoft/MapsSDK-Unity#131]
**Bug Description**
When setting (from script) the MapRenderer.Center to a LatLon that is too far from where it currently is, an error is thrown.

**To Reproduce**
The first of those two coroutines throws an exception, while the second works.

```
IEnumerator GoRandomMapLocation()
{
        yield return m_BingMapRenderer.WaitForLoad();
        m_BingMapRenderer.Center = new LatLon(Random.Range(-90f, 90f), Random.Range(-180f, 180f));
}

IEnumerator GoNotTooFarMapLocation()
{
        yield return m_BingMapRenderer.WaitForLoad();
        m_BingMapRenderer.Center = new LatLon(m_BingMapRenderer.Center.LatitudeInDegrees + 0.0001f, m_BingMapRenderer.Center.LongitudeInDegrees + 0.0001f);
}
```

**Full Console Error Message**
```
System.ArgumentNullException: Buffer cannot be null.
Parameter name: buffer
  at System.IO.MemoryStream..ctor (System.Byte[] buffer, System.Boolean writable) [0x00018] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.IO.MemoryStream..ctor (System.Byte[] buffer) [0x00000] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at (wrapper remoting-invoke-with-check) System.IO.MemoryStream..ctor(byte[])
  at =a8d=.=cc0= (System.Threading.CancellationToken ) [0x00000] in <0a8a7f80b84c4b0c9f79b1816f1316a4>:0 
  at =aad9=+=ba2B=.MoveNext () [0x00124] in <0a8a7f80b84c4b0c9f79b1816f1316a4>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter[TResult].GetResult () [0x00000] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at =a4a=+=baEa=.MoveNext () [0x00272] in <0a8a7f80b84c4b0c9f79b1816f1316a4>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter[TResult].GetResult () [0x00000] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at =aDe=`2+=baBD=`2[=eb2=,=eb3=].MoveNext () [0x00196] in <0a8a7f80b84c4b0c9f79b1816f1316a4>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter[TResult].GetResult () [0x00000] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at =a4a=+=baEE=.MoveNext () [0x0007d] in <0a8a7f80b84c4b0c9f79b1816f1316a4>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter[TResult].GetResult () [0x00000] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at =a4a=+=baEF=.MoveNext () [0x00106] in <0a8a7f80b84c4b0c9f79b1816f1316a4>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter[TResult].GetResult () [0x00000] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at =a4a=+=baEb=.MoveNext () [0x00078] in <0a8a7f80b84c4b0c9f79b1816f1316a4>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter[TResult].GetResult () [0x00000] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at =a52=+=bae7=.MoveNext () [0x000f1] in <0a8a7f80b84c4b0c9f79b1816f1316a4>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter[TResult].GetResult () [0x00000] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at =a52=+=bae4=.MoveNext () [0x00086] in <0a8a7f80b84c4b0c9f79b1816f1316a4>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter[TResult].GetResult () [0x00000] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at =a9F=+=ba0C=.MoveNext () [0x00165] in <0a8a7f80b84c4b0c9f79b1816f1316a4>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter[TResult].GetResult () [0x00000] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at =aBa=+=b91=.MoveNext () [0x0009d] in <0a8a7f80b84c4b0c9f79b1816f1316a4>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter[TResult].GetResult () [0x00000] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at =aBf=+=baAC=`1[=eB8=].MoveNext () [0x00171] in <0a8a7f80b84c4b0c9f79b1816f1316a4>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1+ConfiguredTaskAwaiter[TResult].GetResult () [0x00000] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at =aBf=+=baAB=.MoveNext () [0x0011c] in <0a8a7f80b84c4b0c9f79b1816f1316a4>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable+ConfiguredTaskAwaiter.GetResult () [0x00000] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at =aBf=+=baAa=.MoveNext () [0x0024a] in <0a8a7f80b84c4b0c9f79b1816f1316a4>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) [0x0003e] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) [0x00028] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) [0x00008] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at System.Runtime.CompilerServices.ConfiguredTaskAwaitable+ConfiguredTaskAwaiter.GetResult () [0x00000] in <695d1cc93cca45069c528c15c9fdd749>:0 
  at =aBf=+=baAA=.MoveNext () [0x0007f] in <0a8a7f80b84c4b0c9f79b1816f1316a4>:0 
UnityEngine.Debug:LogError (object)
=aBf=/=baAA=:MoveNext ()
System.Runtime.CompilerServices.AsyncTaskMethodBuilder:SetException (System.Exception)
=aBf=/=baAa=:MoveNext ()
System.Runtime.CompilerServices.AsyncTaskMethodBuilder:SetException (System.Exception)
=aBf=/=baAB=:MoveNext ()
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1<System.Collections.Generic.Dictionary`2<=aBa=, System.Collections.Generic.List`1<Microsoft.Geospatial.TileId>>>:SetException (System.Exception)
=aBf=/=baAC=`1<=aBa=>:MoveNext ()
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1<bool>:SetException (System.Exception)
=aBa=/=b91=:MoveNext ()
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1<System.Nullable`1<=aBA=>>:SetException (System.Exception)
=a9F=/=ba0C=:MoveNext ()
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1<System.Nullable`1<=a55=>>:SetException (System.Exception)
=a52=/=bae4=:MoveNext ()
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1<System.Nullable`1<=a4D=>>:SetException (System.Exception)
=a52=/=bae7=:MoveNext ()
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1<System.Nullable`1<=a4D=>>:SetException (System.Exception)
=a4a=/=baEb=:MoveNext ()
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1<=a31=>:SetException (System.Exception)
=a4a=/=baEF=:MoveNext ()
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1<=a31=>:SetException (System.Exception)
=a4a=/=baEE=:MoveNext ()
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1<=a31=>:SetException (System.Exception)
=aDe=`2/=baBD=`2<Microsoft.Geospatial.TileId, =a31=>:MoveNext ()
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1<=aaBB=`2<Microsoft.Geospatial.TileId, =a31=>>:SetException (System.Exception)
=a4a=/=baEa=:MoveNext ()
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1<System.IO.Stream>:SetException (System.Exception)
=aad9=/=ba2B=:MoveNext ()
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1<=a47=>:SetResult (=a47=)
=aa19=/=ba4a=`1<object>:MoveNext ()
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1<=a47=>:SetResult (=a47=)
=aaE9=/=ba2D=`1<object>:MoveNext ()
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1<=a47=>:SetResult (=a47=)
=a8E=/=baf3=`1<object>:MoveNext ()
System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1<=a8d=>:SetResult (=a8d=)
=a8E=/=baf5=`1<object>:MoveNext ()
System.Runtime.Compil

```

**Comment by FredMoreau** [comment]
This actually happens with Unity 2021.1. Not with 2020.3 LTS

**Comment by kircher1** [comment]
Fixed in 0.11.1
